lib/web/machineid.go

```diff
@@ -223,54 +223,21 @@ func (h *Handler) getBot(w http.ResponseWriter, r *http.Request, p httprouter.Pa
 // updateBot updates a bot with provided roles. The only supported change via this endpoint today is roles.
 // TODO(nicholasmarais1158) DELETE IN v20.0.0 - replaced by updateBotV2
 // MUST delete with related code found in `web/packages/teleport/src/services/bot/bot.ts`
-func (h *Handler) updateBot(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, cluster reversetunnelclient.Cluster) (any, error) {
-	var request updateBotRequest
-	if err := httplib.ReadResourceJSON(r, &request); err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	botName := p.ByName("name")
-	if botName == "" {
-		return nil, trace.BadParameter("empty name")
-	}
-
-	clt, err := sctx.GetUserClient(r.Context(), cluster)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	mask, err := fieldmaskpb.New(&machineidv1.Bot{}, "spec.roles")
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	updated, err := clt.BotServiceClient().UpdateBot(r.Context(), &machineidv1.UpdateBotRequest{
-		UpdateMask: mask,
-		Bot: &machineidv1.Bot{
-			Kind:    types.KindBot,
-			Version: types.V1,
-			Metadata: &headerv1.Metadata{
-				Name: botName,
-			},
-			Spec: &machineidv1.BotSpec{
-				Roles: request.Roles,
-			},
-		},
-	})
-	if err != nil {
-		return nil, trace.Wrap(err, "unable to find existing bot")
-	}
-
-	return updated, nil
-}
-
-type updateBotRequest struct {
-	Roles []string `json:"roles"`
+func (h *Handler) updateBotV1(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, cluster reversetunnelclient.Cluster) (any, error) {
+	return h.updateBotV3(w, r, p, sctx, cluster)
 }
 
 // updateBotV2 updates a bot with provided roles, traits and max_session_ttl.
+// TODO(nicholasmarais1158) DELETE IN v20.0.0 - replaced by updateBotV2
+// MUST delete with related code found in `web/packages/teleport/src/services/bot/bot.ts`
 func (h *Handler) updateBotV2(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, cluster reversetunnelclient.Cluster) (any, error) {
-	var request updateBotRequestV2
+	return h.updateBotV3(w, r, p, sctx, cluster)
+}
+
+// updateBot updates a bot with provided roles, traits, max_session_ttl and
+// description.
+func (h *Handler) updateBotV3(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, cluster reversetunnelclient.Cluster) (any, error) {
```

Comment:
I think the implementation of this should be `updateBot` that takes roles, traits, etc, and each versioned endpoint handler should unmarshal their respective versioned struct and pass through defaults for the values that aren't supported

As it stands, there's no indicator in the code what the difference between the 3 different versions are, and all 3 will accept the latest version with the description. v1 and v2 are existing contracts and shouldn't work with a different request body. Having each versioned handler with their own request struct will make it obvious what the differences are between them

I understand it's not great when you're just adding new optional fields, but existing APIs shouldn't be edited after release, regardless of what the change is